UnitsNet/IValueQuantity.cs

```diff
@@ -48,5 +48,24 @@ public interface IValueQuantity<out TValueType> : IQuantity
         /// <param name="unitSystem">The <see cref="UnitSystem"/> to convert the quantity value to.</param>
         /// <returns>The converted value.</returns>
         new TValueType As(UnitSystem unitSystem);
+
+        /// <summary>
+        ///     Converts this <see cref="IValueQuantity{TValueType}"/> to an <see cref="IValueQuantity{TValueType}"/> in the given <paramref name="unit"/>.
+        /// </summary>
+        /// <param name="unit">
+        ///     The unit <see cref="Enum"/> value. The <see cref="Enum"/> must be compatible with the units of the <see cref="IValueQuantity{TValueType}"/>.
+        ///     For example, if the <see cref="IValueQuantity{TValueType}"/> is a <see cref="Length"/>, you should provide a <see cref="LengthUnit"/> value.
+        /// </param>
+        /// <exception cref="NotImplementedException">Conversion was not possible from this <see cref="IValueQuantity{TValueType}"/> to <paramref name="unit"/>.</exception>
```

Comment:
I'm thinking NotSupportedException or InvalidOperationException instead, since units can be configured at runtime so it is generally not due to a missing implementation, but rather a mistake by the caller (wrong enum) or missing configuration for custom units.

This would have to change in other places too, could be a separate PR though.

Reply:
@angularsen sounds good. I'll keep as-is for this PR then (for consistency) and we can take a pass for another PR.

I assume this would be a v6 issue since it would be a breaking behavior change.